src/Build/BackEnd/BuildManager/BuildManager.cs

```diff
@@ -1725,68 +1725,11 @@ private void ExecuteGraphBuildScheduler(GraphBuildSubmission submission)
                 var cacheServiceTask = Task.Run(() => SearchAndInitializeProjectCachePluginFromGraph(projectGraph));
 
                 IReadOnlyDictionary<ProjectGraphNode, ImmutableList<string>> targetLists = targetListTask.Result;
-                using var cacheService = cacheServiceTask.Result;
 
-                var waitHandle = new AutoResetEvent(true);
-                var graphBuildStateLock = new object();
-
-                var blockedNodes = new HashSet<ProjectGraphNode>(projectGraph.ProjectNodes);
-                var finishedNodes = new HashSet<ProjectGraphNode>(projectGraph.ProjectNodes.Count);
-                var buildingNodes = new Dictionary<BuildSubmission, ProjectGraphNode>();
-                var resultsPerNode = new Dictionary<ProjectGraphNode, BuildResult>(projectGraph.ProjectNodes.Count);
-
-                while (blockedNodes.Count > 0 || buildingNodes.Count > 0)
+                Dictionary<ProjectGraphNode, BuildResult> resultsPerNode;
+                using (cacheServiceTask.Result)
```

Comment:
Can you explain why this changes the ordering versus when the graph build returns?

Reply:
using variables get disposed at the end of the block they're in. In this case the end of the block is **after** the last line in the block whish is the call to ReportResultsToSubmission, but the cache needs to be disposed (in order to surface any issues in EndBuildAsync) **before** the call to ReportResultsToSubmission.